tests/integration/src/test/java/org/apache/pulsar/tests/integration/topologies/PulsarCluster.java

```diff
@@ -358,39 +352,76 @@ public void startPrestoWorker() {
     public void startPrestoWorker(String offloadDriver, String offloadProperties) {
         log.info("[startPrestoWorker] offloadDriver: {}, offloadProperties: {}", offloadDriver, offloadProperties);
         if (null == prestoWorkerContainer) {
-            prestoWorkerContainer = new PrestoWorkerContainer(clusterName, PrestoWorkerContainer.NAME)
-                .withNetwork(network)
-                .withNetworkAliases(PrestoWorkerContainer.NAME)
-                .withEnv("clusterName", clusterName)
-                .withEnv("zkServers", ZKContainer.NAME)
-                .withEnv("zookeeperServers", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
-                .withEnv("pulsar.zookeeper-uri", ZKContainer.NAME + ":" + ZKContainer.ZK_PORT)
-                .withEnv("pulsar.broker-service-url", "http://pulsar-broker-0:8080");
-            if (spec.queryLastMessage) {
-                prestoWorkerContainer.withEnv("pulsar.bookkeeper-use-v2-protocol", "false")
-                    .withEnv("pulsar.bookkeeper-explicit-interval", "10");
-            }
-            if (offloadDriver != null && offloadProperties != null) {
-                log.info("[startPrestoWorker] set offload env offloadDriver: {}, offloadProperties: {}",
-                        offloadDriver, offloadProperties);
-                prestoWorkerContainer.withEnv("PULSAR_PREFIX_pulsar.managed-ledger-offload-driver", offloadDriver);
-                prestoWorkerContainer.withEnv("PULSAR_PREFIX_pulsar.offloader-properties", offloadProperties);
-                prestoWorkerContainer.withEnv("PULSAR_PREFIX_pulsar.offloaders-directory", "/pulsar/offloaders");
-                // used in s3 tests
-                prestoWorkerContainer.withEnv("AWS_ACCESS_KEY_ID", "accesskey");
-                prestoWorkerContainer.withEnv("AWS_SECRET_KEY", "secretkey");
-            }
+            prestoWorkerContainer = buildPrestoWorkerContainer(
+                    PrestoWorkerContainer.NAME, true, offloadDriver, offloadProperties);
         }
-        log.info("[startPrestoWorker] Starting Presto Worker");
         prestoWorkerContainer.start();
+        log.info("[{}] Presto coordinator start finished.", prestoWorkerContainer.getContainerName());
     }
 
     public void stopPrestoWorker() {
         if (null != prestoWorkerContainer) {
             prestoWorkerContainer.stop();
-            log.info("Stopped Presto Worker");
+            log.info("Stopped presto coordinator.");
             prestoWorkerContainer = null;
         }
+        if (sqlFollowWorkerContainers.size() > 0) {
```

Comment:
In the start process, we start prestoWorkerContainer firstly, we may need close it lastly?
```
        pulsarCluster.startPrestoWorker(OFFLOAD_DRIVER, offloadProperties);
        pulsarCluster.startPrestoFollowWorkers(2, OFFLOAD_DRIVER, offloadProperties);
```